omron_controller:
  feedback:
    use_open_loop: {
      type: bool,
      default_value: true,
      description: "open or closed velocity loop",
    }
    kp: {
      type: double,
      default_value: 1.0,
      description: "P-gain",
      validation: {
        gt_eq<>: 0.0
      }
    }
  cmd_vel_topic: {
    type: string,
    default_value: "command",
    description: "Topic to read when not in chain",
    validation: {
      not_empty<>: null
    }
  }
  tf:
    use_tf: {
      type: bool,
      default_value: true,
      description: "broadcast pose through tf"
    }
    from: {
      type: string,
      default_value: "",
      description: "transform from ...",
      validation: {
        not_empty<>: null
      }
    }
    to: {
      type: string,
      default_value: "",
      description: "transform to ...",
      validation: {
        not_empty<>: null
      }
    }
    odom: {
      type: string,
      default_value: "",
      description: "... passing through odometry ...",
      validation: {
        not_empty<>: null
      }
    }
  interfaces:
    velocity:
      prefix: {
        type: string,
        default_value: "omron_base",
        description: "Prefix of the interfaces"
      }
      command: {
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }
      states: {
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }
    pose:
      prefix: {
        type: string,
        default_value: "omron_base/pose",
        description: "Prefix of the interfaces"
      }
      command: { # Only for fake control
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }
      states: {
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }
    reference:
      prefix: {
        type: string,
        default_value: "omron_base/reference",
        description: "Prefix of the interfaces"
      }
      command: {
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }

